File: packages/java/parser-jvm-plugin-nonnull/src/test/java/dev/hilla/parser/plugins/nonnull/basic/BasicEndpoint.java
package dev.hilla.parser.plugins.nonnull.basic;

import java.util.List;
import java.util.Map;
import java.util.Optional;

@Endpoint
public class BasicEndpoint {
    @Nonnull
    public Map<String, @Nonnull List<@Nonnull ExtendedNonNullableModel>> complexType(
            @Nonnull Map<String, @Nonnull List<@Nonnull ExtendedNonNullableModel>> map) {
        return map;
    }

    @javax.annotation.Nonnull
    public List<@Nonnull String> mixedAnnotations(@Nonnull String parameter) {
        return List.of(parameter);
    }

    @javax.annotation.Nonnull
    public String nonTypeAnnotation(
            @javax.annotation.Nonnull String nonTypeParameter) {
        return nonTypeParameter;
    }

    public String nullableType(String nullableParameter) {
        return nullableParameter;
    }

    @Nonnull
    public String simpleType(@Nonnull String str) {
        return str;
    }

    public List<@Nonnull ? extends String> typeArgumentWildcard(
            List<@Nonnull ? extends String> list) {
        return list;
    }

    public <@Nonnull T extends String> List<T> typeParameter(List<T> list) {
        return list;
    }

    public List<@Nonnull NonNullableModel> typeWithTypeArgument(
            List<@Nonnull NonNullableModel> list) {
        return list;
    }

    @Nonnull
    public Optional<@Nonnull String> optional(
            @Nonnull Optional<@Nonnull String> opt) {
        return opt;
    }

    @Nonnull
    public NonNullableFieldModel nonNullableFieldModel(
            @Nonnull NonNullableFieldModel nonNullableFieldModel) {
        return nonNullableFieldModel;
    }

    public static class ExtendedNonNullableModel extends NonNullableModel {
        @javax.annotation.Nonnull
        public List<@Nonnull String> mixedAnnotations;

        @javax.annotation.Nonnull
        public String nonTypeAnnotation;
    }

    public static class NonNullableModel {
        @Nonnull
        public Map<String, @Nonnull List<@Nonnull NonNullableModel>> complexTypeField;
        public String nullableField;
        @Nonnull
        public String protectedField;
        @Nonnull
        public String publicField;
        public List<@Nonnull String> typeWithTypeArgument;
    }

    public static class NonNullableFieldModel {
        @Nonnull
        private List<@Nonnull String> stringList;

        public List<String> getStringList() {
            return this.stringList;
        }

        public void setStringList(List<String> stringList) {
            this.stringList = stringList;
        }
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/basic/snapshots/BasicEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import type ExtendedNonNullableModel from './com/vaadin/hilla/parser/plugins/nonnull/basic/BasicEndpoint/ExtendedNonNullableModel.js';
import type NonNullableFieldModel from './com/vaadin/hilla/parser/plugins/nonnull/basic/BasicEndpoint/NonNullableFieldModel.js';
import type NonNullableModel from './com/vaadin/hilla/parser/plugins/nonnull/basic/BasicEndpoint/NonNullableModel.js';
import client from './connect-client.default.js';

export async function complexType(
  map: Record<string, Array<ExtendedNonNullableModel>>,
  init?: EndpointRequestInit,
): Promise<Record<string, Array<ExtendedNonNullableModel>>> {
  return client.call('BasicEndpoint', 'complexType', { map }, init);
}

export async function mixedAnnotations(parameter: string, init?: EndpointRequestInit): Promise<Array<string>> {
  return client.call('BasicEndpoint', 'mixedAnnotations', { parameter }, init);
}

export async function nonNullableFieldModel(
  nonNullableFieldModel: NonNullableFieldModel,
  init?: EndpointRequestInit,
): Promise<NonNullableFieldModel> {
  return client.call('BasicEndpoint', 'nonNullableFieldModel', { nonNullableFieldModel }, init);
}

export async function nonTypeAnnotation(nonTypeParameter: string, init?: EndpointRequestInit): Promise<string> {
  return client.call('BasicEndpoint', 'nonTypeAnnotation', { nonTypeParameter }, init);
}

export async function nullableType(
  nullableParameter: string | undefined,
  init?: EndpointRequestInit,
): Promise<string | undefined> {
  return client.call('BasicEndpoint', 'nullableType', { nullableParameter }, init);
}

export async function optional(opt: string | undefined, init?: EndpointRequestInit): Promise<string | undefined> {
  return client.call('BasicEndpoint', 'optional', { opt }, init);
}

export async function simpleType(str: string, init?: EndpointRequestInit): Promise<string> {
  return client.call('BasicEndpoint', 'simpleType', { str }, init);
}

export async function typeArgumentWildcard(
  list: Array<string> | undefined,
  init?: EndpointRequestInit,
): Promise<Array<string> | undefined> {
  return client.call('BasicEndpoint', 'typeArgumentWildcard', { list }, init);
}

export async function typeParameter(
  list: Array<string> | undefined,
  init?: EndpointRequestInit,
): Promise<Array<string> | undefined> {
  return client.call('BasicEndpoint', 'typeParameter', { list }, init);
}

export async function typeWithTypeArgument(
  list: Array<NonNullableModel> | undefined,
  init?: EndpointRequestInit,
): Promise<Array<NonNullableModel> | undefined> {
  return client.call('BasicEndpoint', 'typeWithTypeArgument', { list }, init);
}
